pkg/services/cloudmigration/cloudmigrationimpl/cloudmigration.go

```diff
@@ -466,26 +487,36 @@ func (s *Service) CreateSnapshot(ctx context.Context, signedInUser *user.SignedI
 			s.cancelMutex.Unlock()
 		}()
 
-		ctx, cancelFunc := context.WithCancel(context.Background())
+		// Create context out the span context to ensure the trace is propagated
+		asyncCtx := trace.ContextWithSpanContext(context.Background(), span.SpanContext())
+		asyncCtx, asyncSpan := s.tracer.Start(asyncCtx, "CloudMigrationService.CreateSnapshotAsync")
```

Comment:
Isn't the problem that we were using `context.Background` instead of the `ctx` we got when the span was created?

```go
ctx, span := s.tracer.Start(ctx, "CloudMigrationService.CreateSnapshot", trace.WithAttributes(
		attribute.String("sessionUid", sessionUid),
	))
	defer span.End()
```

Reply:
What was the reason for creating a detached context to begin with? Was it because of some possible cancellation due to timeout?
Maybe we can just do this:

```suggestion
		asyncCtx := trace.ContextWithSpanContext(context.WithoutCancel(ctx), span.SpanContext())
		asyncCtx, asyncSpan := s.tracer.Start(asyncCtx, "CloudMigrationService.CreateSnapshotAsync")
```

Reply:
@macabu @PoorlyDefinedBehaviour 

the context of the request is cancelled as soon as you respond back to the client which will abort the asynchronous go routine way too early. 

You always need to be careful with this with execution originated in HTTP requests.

Rule of thumb: 
- Use request context if you want the execution to stop when the request finished o it's aborted. 
- Otherwise, create a new context.

Reply:
> the context of the request is cancelled as soon as you respond back to the client which will abort the asynchronous go routine way too early.

Okay that is what I expected, hence using the `context.WithoutCancel` there but I suppose since you are pulling the span from the previous context when creating a new one, it has a similar behaviour 👍